src/lcm/simulation/simulate.py

```diff
@@ -519,11 +530,11 @@ def _calculate_discrete_argmax(values, choice_axes):
 # ======================================================================================
 
 
-def dict_product(d):
+def dict_product(d: dict[str, Array]) -> tuple[dict[str, Array], int]:
     """Create a product of the entries of a dictionary.
 
     Args:
-        d (dict): Dictionary where all values are arrays.
+        d: Dictionary where all values are arrays, and keys are strings.
```

Comment:
```suggestion
        d: Dictionary.
```
Just repeats the typing info.